lib/Dialect/SV/SVOps.cpp

```diff
@@ -976,6 +977,111 @@ void OrderedOutputOp::build(OpBuilder &builder, OperationState &result,
     body();
 }
 
+//===----------------------------------------------------------------------===//
+// ForOp
+//===----------------------------------------------------------------------===//
+
+void ForOp::build(OpBuilder &builder, OperationState &result,
+                  int64_t lowerBound, int64_t upperBound, int64_t step,
+                  IntegerType type, StringRef name,
+                  llvm::function_ref<void(BlockArgument)> body) {
+  auto lb = builder.create<hw::ConstantOp>(result.location, type, lowerBound);
+  auto ub = builder.create<hw::ConstantOp>(result.location, type, upperBound);
+  auto st = builder.create<hw::ConstantOp>(result.location, type, step);
+  build(builder, result, lb, ub, st, name, body);
```

Comment:
Ah just realized that the bounds and step size are SSA values. In that case `logic` makes sense since we've been using four-value logic everywhere else 🤔

Reply:
Yes, currently sv.for is capable of creating non-synthesizable loops.